ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -783,16 +833,26 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
       -> AnchoredFragment x
          -- ^ The fragment with the new block @b@ as its tip, with the same
          -- anchor as @curChain@.
-      -> Maybe LoELimit
-    computeLoEMaxExtra (LoEEnabled loeFrag) newBlockFrag
-        | rollback > k  = Nothing
-        | otherwise = Just $ LoELimit $ k - rollback
-       where
-         d = Diff.diff newBlockFrag loeFrag
-         rollback = Diff.getRollback d
-
+      -> Maybe (LoE Word64)
+    computeLoEMaxExtra (LoEEnabled loeFrag) newBlockFrag =
+        -- Both fragments are on the same chain
+        if loeSuffixLength == 0 || rollback == 0 then
+          if rollback > k + loeSuffixLength
+            then Nothing
+            else Just $ LoEEnabled $ k + loeSuffixLength - rollback
+        else
+          -- REVIEW: Since we filter out candidates that fork off the LoE later,
+          -- it does not really seem to matter what we return here and we could
+          -- maybe just return @Nothing@ or merge it in the previous case?
```

Comment:
I would leave the code as is. But unless you want other reviewers to give feedback here, I would remove the REVIEW note before merging.

Reply:
```suggestion
```

Reply:
Agreed.